submissions/tavor118sn/task_memory_game/script.js

```diff
@@ -0,0 +1,124 @@
+// General stuff
+
+const GAME_BOARD = document.getElementById('gameBoard');
+let openedCards = [];
+let pairsFound = 0;
+
+const CARDS = [
+  '8-Bit',
+  'Brok',
+  'Colt',
+  'Shelly',
+];
+
+const pairsNumber = CARDS.length;
+
+
+const shuffleArray = (array) => array.sort(function () {
+  return 0.5 - Math.random()
+});
+
+
+// Generate gameboard
+
+const createCard = (elem, index) => {
+
+  const flipCard = document.createElement('div');
+  flipCard.classList.add('flip-card');
+  flipCard.dataset.cardValue = elem;
+
+  const flipInner = document.createElement('div');
+  flipInner.classList.add('flip-card-inner');
+
+  flipInner.innerHTML = `
+    <div class="front">
+      <img class="front-img" src="./img/brawl-stars-main-icon.png" alt="Brawl Icon">
+    </div>
+      <div class="back">
+      <img class="front-img" src="./img/${elem}.png" alt="${elem}">
+    </div>`;
+  flipCard.appendChild(flipInner);
+
+  return flipCard;
+
+};
+
+
+const initGame = () => {
+
+  GAME_BOARD.innerHTML = '';
+  pairsFound = 0;
+
+  let allCards = shuffleArray([...CARDS, ...CARDS]);
+
+  allCards.forEach((elem, index) => {
```

Comment:
You are manipulating DOM in a loop. DOM access operations are expensive. Let's try to do all changes in one operation. No need to append each card one by one

Reply:
Fixed